jerry-core/ecma/base/ecma-gc.c

```diff
@@ -237,130 +370,24 @@ ecma_gc_mark (ecma_object_t *object_p) /**< object to mark from */
 
   if (traverse_properties)
   {
-    for (ecma_property_t *property_p = ecma_get_property_list (object_p), *next_property_p;
-         property_p != NULL;
-         property_p = next_property_p)
+    ecma_property_header_t *prop_iter_p = ecma_get_property_list (object_p);
+
+    while (prop_iter_p != NULL)
     {
-      next_property_p = ECMA_GET_POINTER (ecma_property_t,
-                                          property_p->next_property_p);
+      JERRY_ASSERT (ECMA_PROPERTY_IS_PROPERTY_PAIR (prop_iter_p));
 
-      if (property_p->flags & ECMA_PROPERTY_FLAG_NAMEDDATA)
+      if (prop_iter_p->types[0].type_and_flags != ECMA_PROPERTY_TYPE_DELETED)
```

Comment:
According to the latest measurements the latest changes are already slow down Jerry. I worry that these loops are not that efficient as we would like, and I am reluctant to add more of them.